lib/samples/query_related_features/query_related_features.dart

```diff
@@ -0,0 +1,280 @@
+import 'dart:math';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:flutter/material.dart';
+
+import '../../utils/sample_state_support.dart';
+
+class QueryRelatedFeatures extends StatefulWidget {
+  const QueryRelatedFeatures({super.key});
+
+  @override
+  State<QueryRelatedFeatures> createState() => _QueryRelatedFeaturesState();
+}
+
+class _QueryRelatedFeaturesState extends State<QueryRelatedFeatures>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // A flag for when the settings bottom sheet is visible.
+  var _layerDataVisible = false;
+  // A flag for when the features are loading.
+  var _loadingFeatures = false;
+  // Feature layer for the Alaska National Parks.
+  late final FeatureLayer _alaskaNationalParksLayer;
+  // Feature table for the Alaska National Parks features.
+  late final ServiceFeatureTable _alaskaNationalParksFeaturesTable;
+  // Feature table for the Alaska National Parks Preserves layer.
+  late final ServiceFeatureTable _alaskaNationalParksPreservesTable;
+  // Feature table for the Alaska National Parks Species layer.
+  late final ServiceFeatureTable _alaskaNationalParksSpeciesTable;
+  // The name of the selected park.
+  var _selectedParkName = '';
+  // Lists to store the names of the related features.
+  final _featurePreserves = <String>[];
+  final _featureSpecies = <String>[];
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            // Add a map view to the widget tree and set a controller.
+            ArcGISMapView(
+              controllerProvider: () => _mapViewController,
+              onMapViewReady: onMapViewReady,
+              onTap: onTap,
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+      // Display the bottom sheet when the selected layer data is available.
+      bottomSheet: _layerDataVisible ? buildLayerData(context) : null,
+    );
+  }
+
+  Widget buildLayerData(BuildContext context) {
+    // Display the selected park name and related features.
+    return _loadingFeatures
+        ? const CircularProgressIndicator()
+        : Container(
```

Comment:
This probably needs a close button and a SingleChildScrollView like in manage_bookmarks.dart. I don't see a way to close the sheet once it's open, and I imagine if you rotate to landscape mode, the lists won't always fit.